packages/jest-fake-timers/src/modernFakeTimers.ts

```diff
@@ -119,6 +119,32 @@ export default class FakeTimers {
 
   useRealTimers(): void {
     if (this._fakingTime) {
+      // If user installs a spy on any of fake APIs, the fake implementation gets captured as the original.
+      // This means that after `jest.useRealTimers()` is called, `.mockRestore()` would be restoring wrong implementation.
+      // Therefore all fake APIs which are spied should be restored here.
+      for (const fakeApiKey of this._clock.methods) {
+        let fakeMethod: object;
+
+        switch (fakeApiKey) {
+          case 'hrtime':
+          case 'nextTick':
+            fakeMethod = this._global.process[fakeApiKey];
+            break;
+
+          default:
+            fakeMethod = this._global[fakeApiKey];
+        }
+
+        if (
+          '_isMockFunction' in fakeMethod &&
+          fakeMethod._isMockFunction === true &&
+          'mockRestore' in fakeMethod &&
+          typeof fakeMethod.mockRestore === 'function'
+        ) {
```

Comment:
can we reuse `isMockFunction` or something?

Reply:
For my eye the cleanest solution would be to mark `isMockFunction()` as static method. This is breaking and has to wait for Jest 30. That is why I slowed down with this PR.

Other way could be to export it as a separate function and reuse it in `ModuleMocker` as well.